tests/unit/test_merge_dbs.py

```diff
@@ -0,0 +1,619 @@
+import sqlite3
+from pathlib import Path
+
+import pytest
+
+from rclip.rclip_merge_dbs import RclipDBMerger
+
+
+class TestRclipDBMerger:
+  """Test database merge functionality."""
+
+  def create_test_db(self, db_path: Path, image_count: int, start_id: int = 1) -> None:
+    """Create a test database with sample data."""
+    con = sqlite3.connect(db_path)
+
+    # Create schema matching rclip's
+    con.execute("""
+            CREATE TABLE IF NOT EXISTS images (
+                id INTEGER PRIMARY KEY,
+                deleted BOOLEAN,
+                filepath TEXT NOT NULL UNIQUE,
+                modified_at DATETIME NOT NULL,
+                size INTEGER NOT NULL,
+                vector BLOB NOT NULL,
+                indexing BOOLEAN
+            )
+        """)
+
+    con.execute("CREATE TABLE IF NOT EXISTS db_version (version INTEGER)")
+    con.execute("INSERT INTO db_version(version) VALUES (2)")
```

Comment:
If we are going to move forward with this feature, we should share these SQL queries with the logic that initializes the rclip database to prevent them from going out of sync.